libfwupd/fwupd-remote.c

```diff
@@ -135,13 +137,20 @@ fwupd_remote_set_url (FwupdRemote *self, const gchar *url)
 	/* generate the signature URI too */
 	suffix = fwupd_remote_get_suffix_for_keyring_kind (self->keyring_kind);
 	if (suffix != NULL) {
-		g_autofree gchar *url_asc = g_strconcat (url, suffix, NULL);
-		self->uri_asc = fwupd_remote_build_uri (self, url_asc, NULL);
+		self->metadata_uri_sig = g_strconcat (metadata_uri, suffix, NULL);
+		self->uri_asc = fwupd_remote_build_uri (self, self->metadata_uri_sig, NULL);
 		basename_asc = g_path_get_basename (soup_uri_get_path (self->uri_asc));
 		self->filename_asc = g_strdup_printf ("%s-%s", self->id, basename_asc);
 	}
 }
 
+/* note, this has to be set after MetadataURI */
+static void
```

Comment:
I think it's only important if you're using the now-deprecated fwupd_remote_get_uri() functions.